### PR TITLE
Go linux fixes

### PR DESCRIPTION
**Pull request by hartzell**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### New Formulae Submissions:

- [ ] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?

    * `brew audit ...` blows up on my machine trying to install the rubocop gem.  I'll debug this next.

- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

    * yes

### CODE REVIEW

**Comment by hartzell** [comment]
I'd like to keep plugging away at this but could use some guidance.

I've been trying to get things going on real live linux boxes, but that might be too confusing (I have no control of what's installed where when...).

I'm working on getting the linuxbrew docker images/containers up and running.

**Comment by sjackman** [comment]
Thanks for your contribution, George. I've added the constant `OS::NAME` in ce0f667 just now to make your patch easier. Its value is either `darwin` on Mac OS or `linux` on Linux.

File: Library/Formula/caddy.rb [hunk 1]
```diff
@@ -88,8 +88,13 @@ class Caddy < Formula
 
   def install
     ENV["GOPATH"] = buildpath
-    ENV["GOOS"] = "darwin"
-    ENV["GOARCH"] = MacOS.prefer_64_bit? ? "amd64" : "386"
```
Comment (sjackman) [inline Library/Formula/caddy.rb]: Please leave this line as is.

File: Library/Formula/caddy.rb [hunk 2]
```diff
@@ -88,8 +88,13 @@ class Caddy < Formula
 
   def install
     ENV["GOPATH"] = buildpath
-    ENV["GOOS"] = "darwin"
```
Comment (sjackman) [inline Library/Formula/caddy.rb]: Use
```ruby
ENV["GOOS"] = OS::NAME
```

File: Library/Formula/mercurial.rb [hunk 3]
```diff
@@ -13,6 +13,8 @@ class Mercurial < Formula
     sha256 "53036c632fdfd35d7ccdb85756660e8eeb01437afa6ce68ea35291361bf941ea" => :mavericks
   end
 
+  depends_on "python" if OS.linux?
+
```
Comment (sjackman) [inline Library/Formula/mercurial.rb]: Please post `brew gist-logs mercurial` without this patch applied to report the error message.
Use instead
```ruby
depends_on "python" unless OS.mac?
```

**Comment by sjackman** [comment]
Closing this PR in favour of one PR per formula.

**Pull request closed without merging**